extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/QuarkusAsyncHealthCheckFactory.java

```diff
@@ -50,6 +50,7 @@ public Void call() throws Exception {
     @Override
     public Uni<HealthCheckResponse> callAsync(AsyncHealthCheck asyncHealthCheck) {
         Uni<HealthCheckResponse> healthCheckResponseUni = super.callAsync(asyncHealthCheck);
-        return healthCheckResponseUni.runSubscriptionOn(MutinyHelper.executor(vertx));
+        return healthCheckResponseUni.runSubscriptionOn(MutinyHelper.executor(
+                VertxContext.createNewDuplicatedContext(vertx.getOrCreateContext())));
```

Comment:
I still don't understand why you need a new duplicated context. It means that any local data stored in the processing will be lost.

Reply:
Is it possible to make it work with just this change and not the other one? 

You can create a duplicated context if you want to use your own MDC. Unless there is an MDC in the event loop context, this context won't inherit from anything. If that happens, it would be considered a bug and a significant one.